internal/rbd/qos_test.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2024 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rbd
+
+import (
+	"context"
+	"testing"
+)
+
+func TestSetQOS(t *testing.T) {
+	t.Parallel()
+	ctx := context.TODO()
+
+	tests := map[string]string{
+		baseReadIops:            "2000",
+		baseWriteIops:           "1000",
+		baseReadBytesPerSecond:  "209715200",
+		baseWriteBytesPerSecond: "104857600",
+	}
+
+	rv := rbdVolume{}
+
+	err := rv.SetQOS(ctx, tests)
+	if err != nil {
+		t.Errorf("SetQOS failed: %v", err)
+	}
+	if len(rv.QosParameters) != len(tests) {
+		t.Errorf("SetQOS: rbdVolume.QosParameters size = %d, want %d", len(rv.QosParameters), len(tests))
+	}
+	for k, v := range rv.QosParameters {
+		switch k {
+		case readIopsLimit:
+			if v != tests[baseReadIops] {
+				t.Errorf("SetQOS: %s: %s, want %s", k, v, tests[baseReadIops])
+			}
+		case writeIopsLimit:
+			if v != tests[baseWriteIops] {
+				t.Errorf("SetQOS: %s: %s, want %s", k, v, tests[baseReadIops])
```

Comment:
this should probably be `tests[baseWriteIops]`

Reply:
Yup :-)